### PR TITLE
feat(examples): add MiniCPM5-1B fine-tuning recipes

### LINKED ISSUE #2337: Support MiniCPM5-1B
https://huggingface.co/openbmb/MiniCPM5-1B

### PR DESCRIPTION
**Pull request by Ckcinnabar**
Add SFT and LoRA fine-tuning recipes for openbmb/MiniCPM5-1B on the SQuAD dataset. Update model coverage docs to
  include MiniCPM5.

  Closes #2337

  # What does this PR do ?

  Add fine-tuning recipe YAMLs for MiniCPM5-1B (full SFT and LoRA) and update the model coverage documentation to
  reflect MiniCPM5 support.

  # Changelog

  - Add `examples/llm_finetune/minicpm/minicpm5_1b_squad.yaml`: full SFT recipe for openbmb/MiniCPM5-1B on SQuAD
  - Add `examples/llm_finetune/minicpm/minicpm5_1b_squad_peft.yaml`: LoRA fine-tuning recipe (dim=8, alpha=32) for
  openbmb/MiniCPM5-1B on SQuAD
  - Update `tests/ci_tests/configs/llm_finetune/test_recipes.yml`: register both recipes in CI
  - Update `docs/model-coverage/llm/index.md` and `docs/model-coverage/llm/openbmb/minicpm.md`: add MiniCPM5-1B to
  supported models

  # Before your PR is "Ready for review"

  **Pre checks**:

  - [x] Make sure you read and followed [Contributor
  guidelines](https://github.com/NVIDIA-NeMo/Automodel/blob/main/CONTRIBUTING.md)
  - [x] Did you write any new necessary tests?
  - [x] Did you add or update any necessary documentation?

  # Additional Information

  - Related to #2337

### CODE REVIEW

**Comment by HuiyingLi** [comment]
@Ckcinnabar Could you please attach wandb link/screenshot for both recipes. Thank you!

**Comment by Ckcinnabar** [comment]
@HuiyingLi   
Here are the W&B training logs for both recipes (run on T4 16GB, local_batch_size=2):

  - **LoRA** (100 steps, loss: 8.26 → 0.23, val_loss: 0.18):
  https://wandb.ai/champion1-chen-university-of-florida/minicpm5-finetune/runs/0ii242w7
  - **SFT** (50 steps, loss: 8.26 → 0.18, val_loss: 0.17):
  https://wandb.ai/champion1-chen-university-of-florida/minicpm5-finetune/runs/8gn2fkdk

  Note: SFT run OOM'd at step 51 on T4 (16GB). Reducing `local_batch_size` further or using a higher-VRAM GPU (e.g., A100) will allow it to complete all 100 steps.

**Comment by HuiyingLi** [comment]
Hi @Ckcinnabar , I don't have access to the wandb. Screenshots also works.

File: examples/llm_finetune/minicpm/minicpm5_1b_squad.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+# Uncomment and configure for W&B logging
+# wandb:
+#   project: <your_wandb_project>
+#   entity: <your_wandb_entity>
+#   name: minicpm5_1b_squad
+#   save_dir: <your_wandb_save_dir>
+
+ci:
+  recipe_owner: nemo-automodel-maintainers@nvidia.com
```
Comment (HuiyingLi) [inline examples/llm_finetune/minicpm/minicpm5_1b_squad.yaml]: Can you please put owner as "HuiyingLi" and time as 10min instead of 30min for both configs? Thank you

Comment (Ckcinnabar) [reply examples/llm_finetune/minicpm/minicpm5_1b_squad.yaml]: Hi @HuiyingLi 
Sure, I just pushed it and here are the screenshots.

image
image
image
image
 
Note: `minicpm5_1b_squad` (SFT) ran 50/100 steps before OOM on T4 (16GB). The recipe itself works correctly — reducing `local_batch_size` to 1 or using a higher-VRAM GPU will allow it to complete all 100 steps.

**Comment by Ckcinnabar** [comment]
Hi @HuiyingLi 
Sure, I just pushed it and here are the screenshots.

image
image
image
image
 
Note: `minicpm5_1b_squad` (SFT) ran 50/100 steps before OOM on T4 (16GB). The recipe itself works correctly — reducing `local_batch_size` to 1 or using a higher-VRAM GPU will allow it to complete all 100 steps.

**Comment by HuiyingLi** [comment]
Hi @Ckcinnabar do you mind also sharing a log for hellaswag dataset? Thank you

**Comment by Ckcinnabar** [comment]
Hi @HuiyingLi , I added a HellaSwag LoRA recipe (minicpm5_1b_hellaswag_peft.yaml) and registered it in the CI recipe list and here are the screenshots.

image
image

**Review by HuiyingLi — DISMISSED** [review]
LGTM thank you!

**Comment by Ckcinnabar** [comment]
> LGTM thank you!

Hi @HuiyingLi, it looks like the required checks are still waiting for status to be reported. Could someone help check whether the CI needs to be retriggered? Thanks!

File: docs/model-coverage/llm/openbmb/minicpm.md
Comment (jgerh) [inline docs/model-coverage/llm/openbmb/minicpm.md]:
```suggestion
[MiniCPM](https://github.com/OpenBMB/MiniCPM) is a compact language model series from OpenBMB and Tsinghua University, designed to deliver strong performance at small parameter counts using model merging and continuous training techniques.
```

Comment (jgerh) [inline docs/model-coverage/llm/openbmb/minicpm.md:73]:
```suggestion
**2. Navigate to the examples directory:**

```

Comment (jgerh) [inline docs/model-coverage/llm/openbmb/minicpm.md]: Verify change

```suggestion
automodel --nproc-per-node=8 examples/llm_finetune/minicpm/minicpm5_1b_squad.yaml \
```

Comment (jgerh) [inline docs/model-coverage/llm/openbmb/minicpm.md]:
```suggestion
Refer to the [Installation Guide](../../../guides/installation.md) and [LLM Fine-Tuning Guide](../../../guides/llm/finetune.md).
```

Comment (jgerh) [inline docs/model-coverage/llm/openbmb/minicpm.md]:
```suggestion
Refer to the [LLM Fine-Tuning Guide](../../../guides/llm/finetune.md).
```

Comment (jgerh) [inline docs/model-coverage/llm/openbmb/minicpm.md]:
```suggestion
## Fine-Tune
```

File: docs/model-coverage/llm/index.md
Comment (jgerh) [inline docs/model-coverage/llm/index.md]:
```suggestion
To run LLMs with NeMo AutoModel, make sure you're using NeMo container version [`25.11.00`](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/nemo-automodel?version=25.11.00) or later. If the model you intend to fine-tune requires a newer version of Transformers, you may need to upgrade to the latest version of NeMo AutoModel by running:
```

Comment (jgerh) [inline docs/model-coverage/llm/index.md]:
```suggestion
NeMo AutoModel supports the [AutoModelForCausalLM](https://huggingface.co/transformers/v3.5.1/model_doc/auto.html#automodelforcausallm) in the [Text Generation](https://huggingface.co/models?pipeline_tag=text-generation&sort=trending) category. During preprocessing, it uses `transformers.AutoTokenizer`, which is sufficient for most LLM cases. If your model requires custom text handling, override the tokenizer in your recipe YAML or provide a custom dataset `_target_`. Refer to [LLM datasets](../../guides/llm/dataset.md) and [dataset overview](../../guides/dataset-overview.md).
```

Comment (jgerh) [inline docs/model-coverage/llm/index.md]:
```suggestion
The models listed above can be fine-tuned using NeMo AutoModel. NeMo AutoModel supports two primary fine-tuning approaches:
```

**Review by jgerh — COMMENTED** [review]
Completed tech pubs review of .md files and provided copyedits and suggested revisions

**Comment by Ckcinnabar** [comment]
Hi @HuiyingLi , all checks should be green now — I added Signed-off-by to the commits that were missing it, so DCO passes. Could you re-review and re-enable auto-merge when you have a chance? Thank you!
@jgerh  I've applied all your copyedit suggestions (See → Refer to, "## Fine-Tune", "by running"). Thanks for the review!

**Review by HuiyingLi — APPROVED** [review]
LGTM thank you!

**Review by thomasdhc — APPROVED** [review]
